src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -50,27 +50,67 @@ class DockerBuildContextError(Exception):
 
 
 class DockerVersionContextError(ValueError):
-    pass
+    @classmethod
+    def attribute_error(
+        cls, context: DockerVersionContextValue, attribute: str
+    ) -> DockerVersionContextError:
+        msg = f"The placeholder {attribute!r} is unknown."
+        if context:
+            msg += f' Try with one of: {", ".join(context.keys())}.'
+        return cls(msg)
 
 
 class DockerVersionContextValue(FrozenDict[str, str]):
```

Comment:
It's not clear what the `Version` is in these classnames... they seem to be for generic argument templating. Is that historical?

Reply:
Yeah, I agree. It started out as the context for the `version` field (which was dropped in favour of `image_tags`). But now, it also applies to the `repository` field (or at least a part of it), so I've been contemplating renaming it, just haven't figured out to what yet.

Reply:
Will address that in a follow up PR...

Reply:
#13955